purchases/src/test/java/com/revenuecat/purchases/BasePurchasesTest.kt

```diff
@@ -0,0 +1,355 @@
+import com.revenuecat.purchases.google.toInAppStoreProduct
+import com.revenuecat.purchases.google.toStoreTransaction
+import com.revenuecat.purchases.identity.IdentityManager
+import com.revenuecat.purchases.interfaces.ReceiveCustomerInfoCallback
+import com.revenuecat.purchases.interfaces.UpdatedCustomerInfoListener
+import com.revenuecat.purchases.models.StoreProduct
+import com.revenuecat.purchases.models.StoreTransaction
+import com.revenuecat.purchases.subscriberattributes.SubscriberAttributesManager
+import com.revenuecat.purchases.utils.ONE_OFFERINGS_RESPONSE
+import com.revenuecat.purchases.utils.STUB_PRODUCT_IDENTIFIER
+import com.revenuecat.purchases.utils.SyncDispatcher
+import com.revenuecat.purchases.utils.createMockOneTimeProductDetails
+import com.revenuecat.purchases.utils.stubPurchaseHistoryRecord
+import com.revenuecat.purchases.utils.stubStoreProduct
+import com.revenuecat.purchases.utils.stubSubscriptionOption
+import io.mockk.Runs
+import io.mockk.clearMocks
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.runs
+import io.mockk.slot
+import org.json.JSONObject
+import org.junit.After
+import org.junit.Before
+import org.junit.runner.RunWith
+import org.robolectric.annotation.Config
+
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+open class BasePurchasesTest {
+    protected val mockBillingAbstract: BillingAbstract = mockk()
+    protected val mockBackend: Backend = mockk()
+    protected val mockCache: DeviceCache = mockk()
+    protected val updatedCustomerInfoListener: UpdatedCustomerInfoListener = mockk()
+    private val mockApplication = mockk<Application>(relaxed = true)
+    protected val mockContext = mockk<Context>(relaxed = true).apply {
+        every {
+            applicationContext
+        } returns mockApplication
+    }
+    protected val mockIdentityManager = mockk<IdentityManager>()
+    protected val mockSubscriberAttributesManager = mockk<SubscriberAttributesManager>()
+    internal val mockCustomerInfoHelper = mockk<CustomerInfoHelper>()
+    lateinit var mockOfferingParser: OfferingParser
+    protected val mockDiagnosticsSynchronizer = mockk<DiagnosticsSynchronizer>()
+    protected val mockOfflineEntitlementsManager = mockk<OfflineEntitlementsManager>()
+    internal val mockPostReceiptHelper = mockk<PostReceiptHelper>()
+
+    protected var capturedPurchasesUpdatedListener = slot<BillingAbstract.PurchasesUpdatedListener>()
+    protected var capturedBillingWrapperStateListener = slot<BillingAbstract.StateListener>()
+    private val capturedConsumePurchaseWrapper = slot<StoreTransaction>()
+    private val capturedShouldTryToConsume = slot<Boolean>()
+
+    protected val randomAppUserId = "\$RCAnonymousID:ff68f26e432648369a713849a9f93b58"
+    protected val appUserId = "fakeUserID"
+    protected lateinit var purchases: Purchases
+    protected val mockInfo = mockk<CustomerInfo>()
```

Comment:
I kept here properties that are used in `BasePurchasesTest`, whatever is only used in `PurchasesTest`, meaning it's not needed for setup and teardown is kept in `Purchases.kt`

Whatever is in this new class has been directly copied from `Purchases.kt`